File: API/API/Models/MessageModel.fs
namespace API.MessageModel

open System
open System.Collections.Concurrent


type Message = {
    Id: Guid
    Date: DateTime
    Sender: string 
    Message: string
}

type MessageRepository = {
    Add : Message -> Option<Message>
    GetAll: unit -> seq<Message>    
}


module MessageRepositoryDb = 

    let private messages = new ConcurrentDictionary<Guid, Message>()

    let add msg = 
        let msgCopy = { msg with Id = Guid.NewGuid() }
        if messages.TryAdd(msgCopy.Id, msgCopy) then
            Some(msgCopy)
        else
            None

    let getAll () = 
        messages.Values |> Seq.cast

File: API/Api-v2/MessageModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Api_v2
{
    public class MessageModel
    {
        public Guid UserId { get; set; }
        public DateTime Date { get; set; }
        public string Message { get; set; }
    }
}
